enterprise/frontend/src/embedding-sdk/components/public/MetabotQuestion/MetabotQuestion.tsx

```diff
@@ -4,18 +4,27 @@ import { t } from "ttag";
 import { InteractiveAdHocQuestion } from "embedding-sdk/components/private/InteractiveAdHocQuestion";
 import { InteractiveQuestionDefaultView } from "embedding-sdk/components/private/InteractiveQuestionDefaultView";
 import { withPublicComponentWrapper } from "embedding-sdk/components/private/PublicComponentWrapper";
-import { Flex, Stack, Text } from "metabase/ui";
+import { Flex, Icon, Paper, Stack, Text } from "metabase/ui";
+import { getErrorMessage } from "metabase-enterprise/metabot/constants";
 
 import { MetabotChatEmbedding } from "./MetabotChatEmbedding";
 import { QuestionDetails } from "./QuestionDetails";
 import { QuestionTitle } from "./QuestionTitle";
 
 const MetabotQuestionInner = () => {
   const [redirectUrl, setRedirectUrl] = useState<string>("");
+  const [messages, setMessages] = useState<string[]>([]);
 
   return (
-    <Flex direction="column" align="center" gap="3rem">
-      <MetabotChatEmbedding onRedirectUrl={setRedirectUrl} />
+    <Flex direction="column" align="center" gap="md">
+      <MetabotChatEmbedding
+        onRedirectUrl={setRedirectUrl}
+        onMessages={setMessages}
+      />
+      {messages.length > 0 &&
```

Comment:
nit: `messages.length > 0` redundant as `[].map(...) === []`

Reply:
that's very fair. I took this right off https://github.com/metabase/metabase/blob/30f7096a45bb8439a9e311b5037460aa2a295439/enterprise/frontend/src/metabase-enterprise/metabot/components/MetabotChat/MetabotChat.tsx#L79

😅 But I see your point. Which is clearly a better version.